app/models/piece.rb

```diff
@@ -1,8 +1,97 @@
-class Piece < ApplicationRecord
-  belongs_to :user
-  belongs_to :game
-
-  def self.types
-    %w(King Queen Rook Bishop Knight Pawn)
-  end
-end
+class Piece < ApplicationRecord
+  belongs_to :user
+  belongs_to :game
+
+  def self.types
+    %w[King Queen Rook Bishop Knight Pawn]
+  end
+
+  def obstructed?(destination_x, destination_y)
+    destination_tests = [horizontal_obstruction?(destination_x, destination_y), vertically_obstruction?(destination_x, destination_y), diagonal_one_obstruction?(destination_x, destination_y), diagonal_two_obstruction?(destination_x, destination_y)]

```

Comment:
more info: http://www.thechrisoshow.com/2009/02/16/using-guard-clauses-in-your-ruby-code/